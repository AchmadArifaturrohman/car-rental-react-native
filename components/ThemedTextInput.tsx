import { TextInput, type TextInputProps, StyleSheet } from 'react-native';

import { useThemeColor } from '@/hooks/useThemeColor';

export type ThemedTextInputProps = TextInputProps & {
  lightColor?: string;
  darkColor?: string;
};

export function ThemedTextInput({
  style,
  lightColor,
  darkColor,
  ...rest
}: ThemedTextInputProps) {
const placeholderColor = useThemeColor({ light: lightColor, dark: darkColor }, 'placeholder');
const textColor = useThemeColor({ light: lightColor, dark: darkColor }, 'text');
  return (
    <TextInput
      style={[
        styles.input,
        style,
        { color: textColor }
      ]}
      placeholderTextColor={placeholderColor}
      {...rest}
    />
  );
}

const styles = StyleSheet.create({
  input: {
    fontSize: 16,
    lineHeight: 24,
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 5,
    padding: 10,
    fontFamily: 'PoppinsRegular'
  },
});
