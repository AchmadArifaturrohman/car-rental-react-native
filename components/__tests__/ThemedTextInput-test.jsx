import * as React from "react";
import renderer from "react-test-renderer";

import { ThemedTextInput } from "../ThemedTextInput";

it(`renders correctly`, () => {
  const tree = renderer
    .create(<ThemedTextInput>Snapshot test!</ThemedTextInput>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
