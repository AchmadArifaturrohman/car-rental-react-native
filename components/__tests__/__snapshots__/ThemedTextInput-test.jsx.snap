// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  placeholderTextColor="#687076"
  style={
    [
      {
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "PoppinsRegular",
        "fontSize": 16,
        "lineHeight": 24,
        "padding": 10,
      },
      undefined,
      {
        "color": "#11181C",
      },
    ]
  }
>
  Snapshot test!
</TextInput>
`;
